Bottom: 6aaac58a9388080b78b7d9addc12475288dacc4a
Top:    30614bfb0507dc49650b92e75312883b451fb7e9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 23:22:13 +0000

Refresh of work-in-progress

---

diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index 9f743fd..6e22326 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -377,12 +377,15 @@ static char *
 batch_prompt_for (const char *msg, const char *default_string)
 {
   static char buf[1024];
+  char *s;
+
   if (default_string)
     printf ("%s [%s] : ", msg, default_string);
   else
     printf ("%s : ", msg);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_string)
+
+  s = fgets (buf, 1024, stdin);
+  if (default_string && (s == NULL || buf[0] == 0))
     strcpy (buf, default_string);
   return buf;
 }
@@ -393,12 +396,15 @@ batch_fileselect (const char *title, const char *descr,
 		  const char *history_tag, int flags)
 {
   static char buf[1024];
+  char *s;
+
   if (default_file)
     printf ("%s [%s] : ", title, default_file);
   else
     printf ("%s : ", title);
-  fgets (buf, 1024, stdin);
-  if (buf[0] == 0 && default_file)
+
+  s = fgets (buf, 1024, stdin);
+  if (default_file && (s == NULL || buf[0] == 0))
     strcpy (buf, default_file);
   return buf;
 }
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index f5f225a..cfc1ed4 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -303,7 +303,7 @@ nogui_prompt_for (const char *msg, const char *default_string)
 
   s = fgets (buf, 1024, stdin);
   if (default_string && (s == NULL || buf[0] == 0))
-    strncpy (buf, default_string);
+    strcpy (buf, default_string);
   return buf;
 }
 
@@ -323,8 +323,8 @@ nogui_fileselect (const char *title, const char *descr,
     printf ("%s : ", title);
 
   s = fgets (buf, 1024, stdin);
-  if (default_string && (s == NULL || buf[0] == 0))
-    strncpy (buf, default_string);
+  if (default_file && (s == NULL || buf[0] == 0))
+    strcpy (buf, default_file);
   return buf;
 }
